applications/matter_bridge/src/bridge_shell.cpp

```diff
@@ -46,6 +66,83 @@ static int add_bridged_device(const struct shell *shell, size_t argc, char **arg
 	} else {
 		shell_fprintf(shell, SHELL_ERROR, "Error: internal\n");
 	}
+}
+
+#ifdef CONFIG_BRIDGED_DEVICE_BLE
+static void bluetooth_device_connected(BLEBridgedDevice *device, bt_gatt_dm *discoveredData, bool discoverySucceeded,
+				       void *context)
+{
+	if (context && device && discoveredData && discoverySucceeded) {
+		BluetoothConnectionContext *ctx = reinterpret_cast<BluetoothConnectionContext *>(context);
+
+		ctx->provider->MatchBleDevice(device);
+		if (ctx->provider->ParseDiscoveredData(discoveredData)) {
+			return;
+		}
+
+		add_device(ctx->shell, ctx->deviceType, ctx->nodeLabel, ctx->provider);
+		chip::Platform::Delete(ctx);
+	}
+}
+#endif /* CONFIG_BRIDGED_DEVICE_BLE */
+
+static int add_bridged_device(const struct shell *shell, size_t argc, char **argv)
+{
+	int deviceType = strtoul(argv[1], NULL, 0);
+	char *nodeLabel = nullptr;
+
+#ifdef CONFIG_BRIDGED_DEVICE_BLE
+	int bleDeviceIndex = strtoul(argv[2], NULL, 0);
+
+	if (argv[3]) {
+		nodeLabel = argv[3];
+	}
+
+	/* The device object can be created once the Bluetooth LE connection will be established. */
+	BluetoothConnectionContext *context = chip::Platform::New<BluetoothConnectionContext>();
+
+	if (!context) {
+		return -ENOMEM;
+	}
+
+	Platform::UniquePtr<BluetoothConnectionContext> contextPtr(context);
+	contextPtr->shell = shell;
+	contextPtr->deviceType = deviceType;
+
+	if (nodeLabel) {
+		strcpy(contextPtr->nodeLabel, nodeLabel);
+	}
+
+	BridgedDeviceDataProvider *provider = create_ble_provider(contextPtr->deviceType);
```

Comment:
We are using a `create_ble_provider()` which crates the concrete version of `BridgedDeviceDataProvider` (`BLEBridgedDeviceProvider`). I suggest to either have one generic factory creating `BridgedDeviceDataProvider` or make the `create_ble_provider` to return concrete type. The latter would require to change the 
`using BleDataProviderFactory = DeviceFactory<BridgedDeviceDataProvider, UpdateAttributeCallback>;` 
into 
`using BleDataProviderFactory = DeviceFactory<BLEBridgedDeviceProvider, UpdateAttributeCallback>;`

Then no casting would be needed below - also if we are sure that the downcast is valid probably the `static_cast` is better then `reinterpret_cast` (we cannot use `dynamic_cast` because of lack of RTTI support).

Reply:
Discussed f2f. Let's keep it as it is. I hope we will find cleaner solution when refactoring the factory.